### PR TITLE
Add Kotlin/Groovy/Scala support

### LINKED ISSUE #9: Scala support

### LINKED ISSUE #165: Add support for Kotlin
Kotlin is this new language created by the JetBeans company. It's a statically typed language for Android development working on the JVM. The language is influency by many languages such as Java, Groovy and C#. It has a lot of potential and I think a language, open source might I've add (https://github.com/JetBrains/kotlin), will be a fine addition for Codewars.

### LINKED ISSUE #175: Add Groovy Support
When I try to create a new Kumite with the language groovy the following message is shown:

> A server error has occurred

Take for example [this Kata](http://www.codewars.com/kumite/574f1b3014b9ad877b0017af?sel=574f1b3014b9ad877b0017af) that is an exact clone of [this](http://www.codewars.com/kumite/56adafd80e62cf13a900004d?sel=56adafd80e62cf13a900004d).

### PR DESCRIPTION
**Pull request by kazk**
### Kotlin 1.1.4

Test Frameworks
- JUnit 4 <http://junit.org/junit4/>
- KotlinTest <https://github.com/kotlintest/kotlintest>

Assertion Libraries
- kotlin.test <https://kotlinlang.org/api/latest/kotlin.test/kotlin.test>
- Kluent <https://github.com/MarkusAmshove/Kluent>
- Expekt <https://github.com/winterbe/expekt>

### Groovy 2.4.12

Test Frameworks
- JUnit 4 <http://junit.org/junit4/>
- Spock <http://spockframework.org>

### Scala 2.12.3

Test Frameworks
- JUnit 4 <http://junit.org/junit4/>
- ScalaTest <http://www.scalatest.org/>

---

- [x] Examples
- [x] Tests
- [x] Documentations
- [x] Update README

---

Closes #9 
Closes #165
Closes #175

### CODE REVIEW

File: frameworks/gradle/src/test/kotlin/expektTests.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import com.winterbe.expekt.should
+
+class ExpektTests {
+  @Test
+  fun canUseExpect() {
+    expect(1).to.equal(1)
+  }
+  @Test
+  fun canUseShould() {
+    1.should.equal(1)
```
Comment (OverZealous) [inline frameworks/gradle/src/test/kotlin/expektTests.kt]: Woah, this is amazing. I didn't realize someone replicated Chai for Kotlin!

What's really amazing is [just how clean and readable the code is](https://github.com/winterbe/expekt), even though it's statically and strongly typed. 😃

**Comment by kazk** [comment]
I think this is now ready for review. The first 3 commits should show the basic idea.
I decided to add these 3 languages in single PR because I wanted to see how Gradle worked with multiple languages registered.

**Comment by ice1000** [comment]
When this pr is merged, I will translate a LARGE number of Java Katas into these three languages :smiley:

**Comment by BillOTei** [comment]
if you need help with the scala support, just let me know!

**Comment by ice1000** [comment]
@kazk Kotlin 1.1.4 is published.

**Comment by ice1000** [comment]
Why I didn't see Kotlin/Scala/Groovy in the preview site?

**Pull request merged**